var/spack/repos/builtin/packages/py-h5sh/package.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyH5sh(PythonPackage):
+    """Shell-like environment for HDF5."""
+
+    homepage = "https://h5sh.readthedocs.io/en/"
+    url      = "https://github.com/sethrj/h5sh/archive/v0.1.0.tar.gz"
+
+    maintainers = ['sethrj']
+
+    version('0.1.0', sha256='52d672e3140323e51084a1bfa2d84012e23c640469242381090a3c7af3e00e4f')
+
+    depends_on('python@2.7:2.8,3.4:', type=('build', 'run'))
+    depends_on('py-setuptools', type=('build', 'run'))
```

Comment:
I don't think setuptools is actually used at run-time, is it?

Reply:
It is, actually. The `setup.py` defines an [entry point](https://chriswarrick.com/blog/2014/09/15/python-apps-the-right-way-entry_points-and-scripts/) for the script in order to generate `bin/h5sh`. The script's contents load `pkg_resources`, so if `py-setuptools` is *not* listed as a runtime dependency, the command will fail:
```console
$ h5sh
Traceback (most recent call last):
  File "/rnsdhpc/code/spack/opt/spack2/clang/py-h5sh/mxxgkvn/bin/h5sh", line 6, in <module>
    from pkg_resources import load_entry_point
ModuleNotFoundError: No module named 'pkg_resources'
```